lib/improver/tests/utilities/test_mathematical_operations.py

```diff
@@ -0,0 +1,318 @@
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the convert_distance_into_number_of_grid_cells function from
+ spatial.py."""
+
+import unittest
+
+import numpy as np
+
+import iris
+from iris.tests import IrisTest
+
+from improver.utilities.mathematical_operations import Integration
+from improver.tests.ensemble_calibration.ensemble_calibration.\
+    helper_functions import set_up_temperature_cube
+
+
+def set_up_height_cube(height_points, cube=set_up_temperature_cube()):
+    """Create a height cube using the existing set_up_temperature_cube
+    utility."""
+    ascending = False
+    if np.all(np.diff(height_points) > 0):
+        ascending = True
+
+    cubelist = iris.cube.CubeList([])
+    for height_point in height_points:
+        temp_cube = cube.copy()
+        height_coord = iris.coords.DimCoord(height_point, "height")
+        if ascending:
+            height_coord.attributes = {"positive": "up"}
+        else:
+            height_coord.attributes = {"positive": "down"}
+        temp_cube.add_aux_coord(height_coord)
+        temp_cube = iris.util.new_axis(temp_cube, "height")
+        cubelist.append(temp_cube)
+    return cubelist.concatenate_cube()
+
+
+class Test__init__(IrisTest):
+
+    """Test the init method."""
+
+    def test_raise_exception(self):
+        """Test that an error is raised, if the direction_of_integration
+        is not valid."""
+        coord_name = "height"
+        direction = "sideways"
+        msg = "The specified direction of integration"
+        with self.assertRaisesRegexp(ValueError, msg):
+            Integration(coord_name, direction_of_integration=direction)
+
+
+class Test__repr__(IrisTest):
+
+    """Test the repr method."""
+
+    def test_basic(self):
+        """Test that the __repr__ returns the expected string."""
+        coord_name = "height"
+        result = str(Integration(coord_name))
+        msg = ('<Integration: coord_name_to_integrate: height, '
+               'start_point: None, end_point: None, '
+               'direction_of_integration: negative>')
+        self.assertEqual(result, msg)
+
+
+class Test_ensure_monotonic_increase_in_chosen_direction(IrisTest):
```

Comment:
Add a test for the documented "Raises: ValueError"

Reply:
I've removed the reference to the ValueError from the method, as a ValueError isn't being raised, so this test isn't needed.